### PR TITLE
Add `kubectl` plugin

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add `kubectl` plugin that handles running full kubectl commands
- Add initial flow/boilerplate for handling interactivity
- Extract the Message that is used for `Help` and `Execute` methods for our plugin
- **New**: Add option to filter basic code blocks and plaintext messages using Botkube built-in `--filter` flag.

## TODO

- What to do with our gRPC contract? remove it? deprecate it? or just rename it?
- Change the context.platform to `SupportsInteractivitiy bool`

## Notes

Kubectl is more complex than Helm, so here I only tried to validate the explicitly not supported commands and global flags. Still, some sub-commands flags can be used even if they don't make sens in case of Botkube plugin. Additionally, help for sub-commands is returned directly from kubectl. This can be adjusted in the future but maintaining that can be overkill comparing to the end value.

## Related issue(s)

#841 

Fix https://github.com/kubeshop/botkube/issues/975

## Demo
https://user-images.githubusercontent.com/17568639/217392394-093d8021-15c7-4c2b-a4c7-4888d53e87d5.mov

### CODE REVIEW

**Review by pkosiec — COMMENTED** [review]
Looks good! Just a few small comments.

Disclaimer: Still testing it, this is just a code review 👍

File: cmd/executor/kubectl/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+Kubectl is the Botkube executor plugin that allows you to run the Kubectl CLI commands directly from any communication platform.
+
+## Configuration parameters
+
+The configuration should be specified in the YAML format. Such parameters are supported:
+
+```yaml
+defaultNamespace: "default"
+```
```
[the remark this answers is not in the document]
Comment (mszostok) [reply cmd/executor/kubectl/README.md:11]: Hm.. I think that testing instruction won't be needed here 🤔 I added ones for helm as it's not tested so I wanted to have this scenario somewhere available for RC testing. But once we will implement the RBAC we should replace the built-in kubectl with plugin/kubectl in e2e test. WDYT?

File: pkg/pluginx/command.go [hunk 2]
```diff
@@ -98,6 +98,22 @@ func ExecuteCommandWithEnvs(ctx context.Context, rawCmd string, envs map[string]
...
+
+	if sout != "" && serr != "" {
+		return fmt.Errorf("%s\n%s\n%w", sout, serr, err)
+	}
+
+	if sout != "" {
+		return fmt.Errorf("%s\n%w", sout, err)
+	}
+
+	return fmt.Errorf("%s\n%w", serr, err)
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/pluginx/command.go:114]: I had those headers before but unfortunately it doesn't look good on Slack, I can add them but not sure if user cares about that in chat window 🤔

File: pkg/api/executor/grpc_adapter.go [hunk 3]
```diff
@@ -76,15 +85,29 @@ type grpcClient struct {
...
+		},
+	}
+
+	res, err := p.client.Execute(ctx, grpcInput)
 	if err != nil {
 		return ExecuteOutput{}, err
 	}
+
+	var msg api.Message
+	if len(res.Message) != 0 && string(res.Message) != "" {
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/api/executor/grpc_adapter.go:102]: copy-paste error, as I took it from help, will improve in both places 👍

File: pkg/bot/interactive/testdata/TestInteractiveMessageToMarkdownMultiSelect.golden.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 **Adjust notifications**
+Adjust notifications description
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/bot/interactive/testdata/TestInteractiveMessageToMarkdownMultiSelect.golden.txt:2]: it's just for testing purposes, I added it to make sure that this field is also taken into account

File: pkg/api/executor/grpc_adapter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/api/executor/grpc_adapter.go:102]: copy-paste error, as I took it from help, will improve in both places 👍

**Comment by pkosiec** [comment]
BTW tested and it works flawlessly 👌 🚀

**Review by pkosiec — APPROVED** [review]
🚀

File: pkg/bot/interactive/base_msg.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package interactive
+
+import "github.com/kubeshop/botkube/pkg/api"
+
+type Message struct {
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/bot/interactive/base_msg.go:6]: used `InternalMessage` 👍

File: pkg/api/executor/grpc_adapter.go [hunk 6]
```diff
@@ -10,29 +10,38 @@ import (
...
+	// ExecuteInputContext holds execution context.
+	ExecuteInputContext struct {
+		CommunicationPlatform config.CommPlatformIntegration
+	}
+
 	// ExecuteOutput holds the output of the Execute function.
 	ExecuteOutput struct {
-		// Data represent the output of processing a given input command.
+		// Data represents the output of processing a given input command.
 		Data string
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/api/executor/grpc_adapter.go:42]: I don't know if I understand that correctly, so for now I will just deprecate it and we handle the "removal/migration" in next PR 👍

File: pkg/execute/plugin_executor.go [hunk 7]
```diff
@@ -52,55 +54,114 @@ func (e *PluginExecutor) GetCommandPrefix(args []string) string {
...
+		},
+	}
+
+	// Show Filter Input if command response is more than `lineLimitToShowFilter`
+	allLines := code + plaintext
+	if len(strings.SplitN(allLines, "\n", lineLimitToShowFilter)) == lineLimitToShowFilter {
+		outMsg.PlaintextInputs = append(outMsg.PlaintextInputs,
+			filterInput(cmdCtx.CleanCmd, cmdCtx.BotName))
+	}
+
```
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/execute/plugin_executor.go:183]: done 👌

File: pkg/api/executor/grpc_adapter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mszostok) [reply pkg/api/executor/grpc_adapter.go:102]: unfortunately empty string results in json.Unmarshal error :(

**Comment by mszostok** [comment]
Thanks! 

all suggestion were applied on this commit: https://github.com/kubeshop/botkube/pull/973/commits/2638bbc8997f8ec1a91c34030a734a045c15081b

**Pull request merged**